src/services/platform.ts

```diff
@@ -12,13 +13,18 @@ async function matchTxToPlatform(tx: string): Promise<IPlatform> {
     const to = parsed.to.toLowerCase();
     const data = parsed.data.toLowerCase();
     const platforms = await Platform.find();
-
     for (const platform of platforms) {
-      for (const contract of platform.whitelist) {
-        const contractLower = contract.toLowerCase();
-        const contractNoPrefix = contractLower.substr(2);
-        if (to === contractLower || data.includes(contractNoPrefix)) {
-          return platform;
+      const platformArbitrum = platform.whitelist.ethereum[ChainId.Arbitrum];
```

Comment:
why hardcode this?? you do understand that this is a generic function for any chain? the `chainId` must come from the transaction being checked. you will find it in the `parsed` object. since our new `platform` data structure also includes `assetType`, you will need to get that as a new parameter

Reply:
yes thank you, i was trying to figure out where i can get chainId.